src/parse/parse.c

```diff
@@ -6,64 +6,147 @@
 /*   By: walnaimi <walnaimi@student.hive.fi>        +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/07/26 11:49:16 by walnaimi          #+#    #+#             */
-/*   Updated: 2024/07/26 12:55:48 by walnaimi         ###   ########.fr       */
+/*   Updated: 2024/07/29 22:53:27 by walnaimi         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
-#include "../includes/minishell.h"
-int	tokens_parsing(t_token *token)
+#include "../../includes/minishell.h"
+
+void enter_file(int fd)
 {
-	if (incorrect_syntax(token, PIPE) == FAILURE
-		|| incorrect_syntax(token, RED_OUT) == FAILURE
-		|| incorrect_syntax(token, RED_IN) == FAILURE
-		|| incorrect_syntax(token, HEREDOC) == FAILURE
-		|| incorrect_syntax(token, APPEND) == FAILURE)
-		return (FAILURE);
-	else
-		return (SUCCESS);
-	
+    char buffer[1024];
+    ssize_t bytes_read;
+
+    while (1)
+    {
+        bytes_read = read(STDIN_FILENO, buffer, sizeof(buffer) - 1);
+        if (bytes_read < 0)
+        {
+            perror("read");
+            break;
+        }
+
+        buffer[bytes_read] = '\0'; // Null-terminate the string
+
+        if (strncmp(buffer, "END", 3) == 0 && (buffer[3] == '\n' || buffer[3] == '\0'))
+        {
+            break;
+        }
+
+        if (write(fd, buffer, bytes_read) != bytes_read)
+        {
+            perror("write");
+            break;
+        }
+    }
+
+    close(fd);
 }
 
-int	incorrect_syntax(t_token *token, t_type token_type)
+int handle_output_redirection(t_token *current_token, t_token *next_token)
 {
-	t_token	*head;
-	t_type	next_type;
-	
-	head = token;
-	while (head)
-	{
-		if (head->type == token_type)
-		{
-			if (head->next != NULL)
-			{
-				next_type = head->next->type;
-				if (next_type == COMMAND || next_type == RED_IN
-					|| next_type == PIPE || next_type == RED_OUT
-					|| next_type == HEREDOC || next_type == APPEND
-					|| next_type == FLAG || next_type == UNKNOWN)
-				return (err_pipes(head->value, 2));
-			}
-		}
-		head = head->next;
-	}
-	head = NULL;
-	return (SUCCESS);
+    int fd;
+
+    if (current_token->type == RED_OUT)
+    {
+        fd = open(next_token->value, O_WRONLY | O_CREAT | O_TRUNC, 0664);
+        if (fd < 0)
+        {
+            perror("open");
+            return FAILURE;
+        }
+        enter_file(fd);
```

Comment:
Shouldn't this be for HERE_DOC? If I understand the logic, this will open a prompt back for the user until they type END. This behavior is consistent to HERE_DOC, but HERE_DOC (a.k.a. "<<").

![image](https://github.com/user-attachments/assets/30f81e09-a973-4978-8bf4-e324abffe081)

Reply:
yeah because I don't know how to use our signals exactly in minishell so for now I just use END to close the file, just as I said in my pull request comment

Reply:
alright I see what's happening :3, somehow I thought all of them works like here_doc but without using END and you will have to exit with ctrl+c, I can remove that part, so basically I made 2 here_docs one appends and one just rewrites, I will change so it doesn't enter file